ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/CommandGenerator.scala

```diff
@@ -60,24 +97,137 @@ final class CommandGenerator(
     randomnessProvider.randomNatural(math.pow(10.0, i.toDouble).toInt) == 0
 
   private def createContractCommand(
-      template: String,
+      templateName: String,
+      signatory: Primitive.Party,
       observers: List[Primitive.Party],
       payload: String,
       archive: Boolean,
-  ): Try[Primitive.Party => Command] =
-    (template, archive) match {
-      case ("Foo1", false) => Success(Foo1(_, observers, payload).create.command)
-      case ("Foo2", false) => Success(Foo2(_, observers, payload).create.command)
-      case ("Foo3", false) => Success(Foo3(_, observers, payload).create.command)
-      case ("Foo1", true) =>
-        Success(Foo1(_, observers, payload).createAnd.exerciseArchive(signatory).command)
-      case ("Foo2", true) =>
-        Success(Foo2(_, observers, payload).createAnd.exerciseArchive(signatory).command)
-      case ("Foo3", true) =>
-        Success(Foo3(_, observers, payload).createAnd.exerciseArchive(signatory).command)
-      case invalid => Failure(new RuntimeException(s"Invalid template: $invalid"))
+  ): CreateCmdAndContinuations = {
+    val consumingExercisePayload: Option[String] = config.consumingExercises
+      .flatMap(c =>
+        Option.when(randomnessProvider.randomDouble() <= c.probability)(c.payloadSizeBytes)
+      )
+      .map(randomPayload)
+    val nonconsumingExercisePayload: Seq[String] =
+      config.nonconsumingExercises.fold(Seq.empty[String]) { c =>
+        var f = c.probability.toInt
+        if (randomnessProvider.randomDouble() <= c.probability - f) {
+          f += 1
+        }
+        Seq.fill[String](f)(randomPayload(c.payloadSizeBytes))
+      }
+    val (templateDesc, createCmd) = templateName match {
+      case "Foo1" => (TemplateDescriptor.Foo1, Foo1(signatory, observers, payload).create.command)
+      case "Foo2" => (TemplateDescriptor.Foo2, Foo2(signatory, observers, payload).create.command)
+      case "Foo3" => (TemplateDescriptor.Foo3, Foo3(signatory, observers, payload).create.command)
+      case invalid => sys.error(s"Invalid template: $invalid")
     }
 
+    CreateCmdAndContinuations(
+      createCommand = createCmd,
+      continuationF = {
+        def createCont(cid: ContractId): Seq[Command] = {
+          val nonconsumingExercises = nonconsumingExercisePayload.map { payload =>
+            createExerciseCmd(
+              templateId = templateDesc.templateId,
+              choiceName = templateDesc.nonconsumingChoiceName,
+              argValue = payload,
+            )(cid)
+          }
+          val consumingExerciseO = consumingExercisePayload.fold[Option[Command]](None)(payload =>
+            Some(
+              createExerciseCmd(
+                templateId = templateDesc.templateId,
+                choiceName = templateDesc.consumingChoiceName,
+                argValue = payload,
+              )(cid)
+            )
+          )
+          // NOTE: Archive will be generated only if prior consuming exercise is absent.
```

Comment:
why do we need two archival probabilities? why would it be practical to have empty archive payload and nonempy in other cases?

Reply:
Ok, removed `Archive`